pkg/drivers/sqlite/sqlite.go

```diff
@@ -105,3 +118,76 @@ func setup(db *sql.DB) error {
 
 	return nil
 }
+
+// alterTableIndices drops the given old table indices from the existing
+// kine table (if it exists, and if the indices exists), and then creates
+// the given new ones on the kine table.
+func alterTableIndices(d *generic.Generic) error {
+	if d.LockWrites {
+		d.Lock()
+		defer d.Unlock()
+	}
+
+	// Drop the old indices
+	for _, stmt := range dropIndices {
+		_, err := d.DB.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create the new indices
+	for _, stmt := range createIndices {
+		_, err := d.DB.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkMigrate performs migration from an old key value table to the kine
+// table only if the old key value table exists and migration has not been
+// done already.
+func checkMigrate(ctx context.Context, d *generic.Generic) error {
+	row := d.DB.QueryRowContext(ctx, userVersionSQL)
```

Comment:
All of these changes **must** occur in a transaction. What would happen if you lost connectivity to the server (= the leader) during this migration? Most likely you would end up with a half-made migration which in general is bad.

Reply:
Yeah, this is tricky - the current implementation simply ignores any errors that may occur during the migration of rows from the old table to the new one. We need to add error handling to migration. I saw other places in the source code where execution of a query is retried a number of times until it succeeds before proceeding. Thinking if we could employ a similar measure here? - i.e. try a number of times, and give up if it fails, in which case, abort backend setup?